### DISCUSSION TITLE
Evaluation Metric Error

### DISCUSSION THREAD

**Discussion by ElXPA** [mlflow/mlflow#12946] [Q&A]
Hi guys, just need some help with eval stuff - so I set up a basic eval pipeline to evaluate gpt's responses, I tested to see what a positive expected outcome for eval with respect to semantic similarity would look like and It seems im getting a bad output. 
![Screenshot 2024-08-14 at 17 07 34](https://github.com/user-attachments/assets/231595d7-2b10-4eb8-b9c0-242b7b1a59f1)
In the SS, you can see that the ground truth which is set as the target, is semantically similar to the output of the model but im getting a low semantic similarity score and negative justification. can someone guide me on this. Additional SS of code: 
![Screenshot 2024-08-14 at 17 11 53](https://github.com/user-attachments/assets/4ed06f0b-bc67-4f3c-8e30-5f515dd6e345)
![Screenshot 2024-08-14 at 17 12 14](https://github.com/user-attachments/assets/27316882-23d5-4050-bf1d-d8f1d9f10e1c)

**Comment by B-Step62** [comment]
@ElXPA Could you share the non-truncated output compared with the ground truth? The metric is computed by LLM (default - gpt-4) so there may be some randomness over the result, especially for short sentence like this.

**Reply by ElXPA** [reply]
Hey @B-Step62  thanks for reaching out. I am no longer able to replicate the previous result but please find attached another bad semantic similarity score - included with full output. 
![Screenshot 2024-08-20 at 11 31 20](https://github.com/user-attachments/assets/d88abc89-9e45-4670-939c-0fed84aa8e3b)
![Screenshot 2024-08-20 at 11 31 30](https://github.com/user-attachments/assets/5dab6f18-f89e-40f6-b2ed-70d028422f91)

"data": [
    [
      "whats with the weather recently?",
      "Hello! I'm here and ready to help with any questions you have.",
      "The recent unusual weather patterns could be attributed to climate change, with changing global temperatures leading to more extreme and unpredictable weather events. It's important to closely monitor and study these patterns to understand their impact on the environment and human activities.",
      0.4909846783,
      46,
      0.000139154,
      13.8,
      17.3,
      3,
      "The output has moderate semantic similarity to the provided targets. It captures the general idea of attributing unusual weather patterns to climate change and the importance of monitoring and studying these patterns, but it may lack some specific details and depth compared to the provided targets.",
      4,
      "The output is largely consistent with the ground truths, reflecting the core meaning accurately with only a few minor differences. The response addresses the unusual weather patterns and attributes them to climate change, emphasizing the importance of monitoring and studying these patterns."
    ]
